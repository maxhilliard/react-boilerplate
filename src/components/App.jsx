import React from 'react';
import PropTypes from 'prop-types';

const App = ({ headerContent }) => (
    <div>
        <h1>
            {headerContent}
        </h1>
    </div>
);

App.propTypes = {
    headerContent: PropTypes.string,
};

App.defaultProps = {
    headerContent: '',
};

export default App;
