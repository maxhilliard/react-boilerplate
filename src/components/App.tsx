import * as React from 'react';

// TODO - resolve issues with typescript and css modules
// import * as styles from './app.css';

type AppProps = { headerContent: string; }

const App = (props: AppProps) => (
    <div>
        <h1>
            {props.headerContent}
        </h1>
    </div>
);

// App.propTypes = {
//     headerContent: PropTypes.string,
// };

// App.defaultProps = {
//     headerContent: 'Hello',
// };

export default App;
