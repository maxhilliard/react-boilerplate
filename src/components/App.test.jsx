import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import App from './App';

describe('The app component', () => {
    it('should match snapshot', () => {
        const tree = renderer.create(<App />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should display the headerContent prop as expected', () => {
        const mockHeaderContent = 'Awesome sauce';
        const wrapper = shallow(<App headerContent={mockHeaderContent} />);

        expect(wrapper.find('h1').text()).toEqual(mockHeaderContent);
    });
});
