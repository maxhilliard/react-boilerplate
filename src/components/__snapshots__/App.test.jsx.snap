// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The app component should match snapshot 1`] = `
<div>
  <h1
    className="heading"
  >
    Hello
  </h1>
</div>
`;
